src/coffee/services/base.coffee

```diff
@@ -541,6 +541,15 @@ class BaseService
             headers: response.headers
       else {}
 
+    options = _.omit(@_rest._options, ['access_token'])
+    if options.config
+      options.config = _.omit(options.config, ['client_id', 'client_secret'])
+
+    originalRequest = _.extend(
+      originalRequest,
+      options
```

Comment:
I would name this `originalOptions: options`, just to make it explicit and to keep the options grouped.